packages/puppeteer-core/src/common/bidi/Page.ts

```diff
@@ -313,6 +313,22 @@ export class Page extends PageBase {
     return this.mainFrame().goto(url, options);
   }
 
+  override async reload(
+    options?: WaitForOptions
+  ): Promise<HTTPResponse | null> {
+    const [response] = await Promise.all([
+      this.waitForResponse(response => {
+        return (
+          response.request().isNavigationRequest() &&
+          response.url() === this.url()
+        );
+      }),
+      this.mainFrame().context().reload(options),
```

Comment:
let's use `this.mainFrame().reload()` ?

Reply:
This mistake on my part. The spec clearly defines any BrowsingContext not only the `TopLevel one` aka `Page` as WebDriver.

Reply:
What I said it true, but I think we want to keep the Puppeteer implementation the same for as per the Docs.